Packages/cdms2/Lib/axis.py

```diff
@@ -735,10 +735,22 @@ def designateTime(self, persistent=0, calendar=None):
             if calendar is not None:
                 self.setCalendar(calendar, persistent)
 
+    # For isTime(), keep track of whether each id is for a time axis or not, for better performance.
+    # This dictionary is a class variable (not a member of any particular instance).
+    idtaxis = {}  # id:type where type is 'T' for time, 'O' for other
+
```

Comment:
The point of this is to not have to figure out what the type is - that involves reading from a file.  So the way I would fix it is to use self.idtaxis only if the user does something to enable this feature, e.g. calls a method to initialize it.